### PR TITLE
feat(vscode-web): support hosting on a subpath with `subdomain=false`

### PR DESCRIPTION
**Pull request by Emyrk**
Requires https://github.com/coder/coder/issues/14483

### CODE REVIEW

File: vscode-web/main.tf [hunk 1]
```diff
@@ -160,7 +169,7 @@ resource "coder_app" "vscode-web" {
   display_name = var.display_name
   url          = var.folder == "" ? "http://localhost:${var.port}" : "http://localhost:${var.port}?folder=${var.folder}"
   icon         = "/icon/code.svg"
-  subdomain    = true
+  subdomain    = var.subdomain
```
Comment (stirby) [inline vscode-web/main.tf:172]: Setting subdomain to false breaks the app. I think you'll have to chang the URL format when it's set.

**Review by stirby — CHANGES_REQUESTED** [review]
Breaks in my testing

**Comment by matifali** [comment]
@Emyrk Do you plan to come back to this? A similar approach is used in #316 and #320.

**Comment by Emyrk** [comment]
@matifali I think this has be done for it to work
https://github.com/coder/coder/issues/14483

I see the Juypter accomplishing this by setting the url to the full path, but I thought I tried that and it did not work for this.

**Comment by matifali** [comment]
@Emyrk A Community member successfully ran Jupyter on a subpath by setting the full URL. See #313

**Comment by Emyrk** [comment]
> @Emyrk A Community member successfully ran Jupyter on a subpath by setting the full URL. See #313

I do see that. I think it depends on how urls are constructed within the app. I _am pretty sure_ I tried this with VSCode and it did not. Using the exact same technique. 

If someone has more time to try it, please do.

**Comment by matifali** [comment]
@code-asher @Emyrk I tested with both `subdomain=true|false` and it works. 
Also, I have noticed that skipping the `coder_agent`'s name auto updates the URL with the correct `agent_name`, so it is not necessary.

We can drop `agent_name` from all modules that support `subdomain=false`.

**Review by matifali — APPROVED** [review]

File: vscode-web/main.tf [hunk 2]
```diff
@@ -138,6 +150,9 @@ resource "coder_script" "vscode-web" {
     EXTENSIONS_DIR : var.extensions_dir,
     FOLDER : var.folder,
     AUTO_INSTALL_EXTENSIONS : var.auto_install_extensions,
+    // TODO: This assumes the agent is called "main"
+    SERVER_BASE_PATH : format("/@%s/%s.%s/apps/vscode-web/",
```
[the remark this answers is not in the document]
Comment (matifali) [reply vscode-web/main.tf]: I have noticed that skipping the `coder_agent`'s name auto updates the URL with the correct `agent_name`, so it is unnecessary.

**Pull request merged**